cmd/repo-updater/repoupdater/server.go

```diff
@@ -406,3 +400,87 @@ func (s *Server) handleSchedulePermsSync(w http.ResponseWriter, r *http.Request)
 
 	s.respond(w, http.StatusOK, nil)
 }
+
+func (s *Server) handleQueryExternalServiceNamespaces(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithCancel(r.Context())
+	defer cancel()
+
+	var req protocol.QueryExternalServiceNamespacesArgs
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	extsvc := &types.ExternalService{
+		Kind:   req.Kind,
+		Config: extsvc.NewUnencryptedConfig(req.Config),
+	}
+
+	logger := s.Logger.With(log.String("ExternalServiceKind", req.Kind))
+
+	genericSourcer := s.NewGenericSourcer(logger)
+	genericSrc, err := genericSourcer(ctx, extsvc)
+	if err != nil {
+		logger.Error("server.query-external-service-namespaces", log.Error(err))
+		return
+	}
+
+	var result *protocol.QueryExternalServiceNamespacesResult
+	if err = genericSrc.CheckConnection(ctx); err != nil {
+		result = &protocol.QueryExternalServiceNamespacesResult{Error: err.Error()}
+		s.respond(w, http.StatusUnauthorized, result)
+		return
+	}
+
+	var (
+		discoverableSrc repos.DiscoverableSource
+		discoverable    bool
+	)
+
+	if discoverableSrc, discoverable = genericSrc.(repos.DiscoverableSource); !discoverable {
+		result = &protocol.QueryExternalServiceNamespacesResult{Error: repos.UnimplementedDiscoverySource}
+		s.respond(w, http.StatusNotImplemented, result)
+		return
+	}
```

Comment:
You won't need the var declaration if you split the `if`:

```suggestion
		discoverableSrc, ok := genericSrc.(repos.DiscoverableSource)
		if !ok{
				result = &protocol.QueryExternalServiceNamespacesResult{Error: repos.UnimplementedDiscoverySource}
				s.respond(w, http.StatusNotImplemented, result)
				return
	}
```


Also in idiomatic Go code we use the "comma, ok" convention for such checks where the variable is named `ok` itself.